core/src/desul/atomics/CUDA.hpp

```diff
@@ -9,11 +9,20 @@ SPDX-License-Identifier: (BSD-3-Clause)
 #define DESUL_ATOMICS_CUDA_HPP_
 
 #ifdef DESUL_HAVE_CUDA_ATOMICS
-#if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__>=700)) || !defined(__NVCC__)
+// When building with clang we need to include the device functions always
+// since clang must see a consistent overload set in both device and host compilation
+// but that means we need to know on the host what to make visible, i.e. we need
+// a host side compile knowledge of architecture.
+// We simply can say DESUL proper doesn't support clang CUDA build pre Volta,
+// Kokkos has that knowledge and so I use it here, allowing in Kokkos to use
+// clang with pre Volta as CUDA compiler
+#if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__>=700)) || \
+    (!defined(__NVCC__) && !defined(KOKKOS_ARCH_KEPLER) && !defined(KOKKOS_ARCH_MAXWELL) && !defined(KOKKOS_ARCH_PASCAL))
```

Comment:
Just a note it would be helpful to add a macro to distinguish pre-Volta GPUs from Volta onward, or perhaps the Cuda arch (could also be used in KokkosKernels and Trilinos)? Cross-reference
https://github.com/kokkos/kokkos-kernels/issues/765 
https://github.com/trilinos/Trilinos/issues/9155#issuecomment-870912161